File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/pool/DeletePoolTest.java
package org.hyperledger.indy.sdk.pool;

import org.hyperledger.indy.sdk.ErrorCode;
import org.hyperledger.indy.sdk.ErrorCodeMatcher;
import org.hyperledger.indy.sdk.IndyException;
import org.hyperledger.indy.sdk.IndyIntegrationTest;
import org.hyperledger.indy.sdk.utils.PoolUtils;
import org.junit.Test;

import java.io.IOException;
import java.util.concurrent.ExecutionException;

import static org.junit.Assert.assertNotNull;

public class DeletePoolTest extends IndyIntegrationTest {

	@Test
	public void testDeletePoolWorks() throws InterruptedException, ExecutionException, IndyException, IOException {
		String poolName = PoolUtils.createPoolLedgerConfig();
		Pool.deletePoolLedgerConfig(poolName).get();
	}

	@Test
	public void testDeletePoolWorksForOpened() throws InterruptedException, ExecutionException, IndyException, IOException {
		thrown.expectCause(new ErrorCodeMatcher(ErrorCode.CommonInvalidState));

		String poolName = PoolUtils.createPoolLedgerConfig();
		Pool pool = Pool.openPoolLedger(poolName, null).get();
		assertNotNull(pool);
		openedPools.add(pool);
		Pool.deletePoolLedgerConfig(poolName).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/Wrapper/PoolTests/DeletePoolTest.cs
﻿using Indy.Sdk.Dotnet.Wrapper;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Indy.Sdk.Dotnet.Test.Wrapper.PoolTests
{
    [TestClass]
    public class DeletePoolTest : IndyIntegrationTestBase
    {
        [TestMethod]
        public async Task TestDeletePoolWorks()
        {
            var poolName = PoolUtils.CreatePoolLedgerConfig();
            await Pool.DeletePoolLedgerConfigAsync(poolName);
        }

        [TestMethod]
        public async Task TestDeletePoolWorksForOpened()
        {
            var poolName = PoolUtils.CreatePoolLedgerConfig();
            var pool = await Pool.OpenPoolLedgerAsync(poolName, null);

            Assert.IsNotNull(pool);
            _openedPools.Add(pool);

            var ex = await Assert.ThrowsExceptionAsync<IndyException>(() =>
                Pool.DeletePoolLedgerConfigAsync(poolName)
            );

            Assert.AreEqual(ErrorCode.CommonInvalidState, ex.ErrorCode);
        }        
    }
}
